app/src/org/commcare/dalvik/activities/EntityMapActivity.java

```diff
@@ -1,288 +1,186 @@
+                        } else {
+                            entityLocations.add(new Pair<>(entity, location));
                         }
                     }
                 }
             }
         }
-        
-        Log.d(TAG, "Loaded. " + legit +" addresses discovered, " + bogus + " could not be located");
-
-        if (legit != 0 && mEntityOverlay.getCenter() != null) {
-            map.getController().animateTo(mEntityOverlay.getCenter());
-        } else if(location != null) {
-            int lat = (int) (location.getLatitude() * 1E6);
-            int lng = (int) (location.getLongitude() * 1E6);
-            GeoPoint point = new GeoPoint(lat, lng);
-            map.getController().animateTo(point);
-        }
-        
-        map.getOverlays().add(mMyLocationOverlay);
-        //The overlay crashes out if you try to draw it and it's empty,
-        //so only add it if we found something.
-        if(mEntityOverlay.size() > 0) {
-            map.getOverlays().add(mEntityOverlay);
-        }
-        map.getController().setZoom(18);
-        map.setClickable(true);
-        map.setEnabled(true);
-        Log.d(TAG, "Done loading");
+        Log.d(TAG, "Loaded. " + entityLocations.size() +" addresses discovered, " + bogusAddresses + " could not be located");
     }
-    
-    private String getLicenseKey() {
-        //If there's a defined debug key in the local environment, use that.
-        int debugId = this.getResources().getIdentifier("maps_api_key_debug","string", this.getPackageName());
-        if(debugId == 0) { 
-            return BuildConfig.MAPS_API_KEY;
+
+    private LatLng getLatLngFromAddress(String address) {
+        LatLng location = null;
+        try {
+            GeoPointData data = new GeoPointData().cast(new UncastData(address));
+            if(data != null) {
+                location = new LatLng(data.getLatitude(), data.getLongitude());
+            }
+        } catch(Exception ex) {
         }
-        return this.getString(debugId);
-    }
 
-    private EvaluationContext entityContext;
+        // Geocaching code removed
 
-    private EvaluationContext getEC() {
-        if(entityContext == null) {
-            entityContext = session.getEvaluationContext(getInstanceInit());
-        }
-        return entityContext;
-    }
-    
-    private AndroidInstanceInitializer getInstanceInit() {
-        return new AndroidInstanceInitializer(session);
+        return location;
     }
 
     @Override
-    protected void onStop() {
-        super.onStop();
-        mMyLocationOverlay.disableCompass(); 
-        mMyLocationOverlay.disableMyLocation();
+    public void onMapReady(final GoogleMap map) {
+        mMap = map;
+
+        // Find bounding region of markers
+        LatLngBounds.Builder builder = new LatLngBounds.Builder();
+        for (Pair<Entity<TreeReference>, LatLng> entityLocation: entityLocations) {
+            Marker marker = map.addMarker(new MarkerOptions()

```

Comment:
I would use `mMap` instead of `map` inside of this method. Doing so makes it clearer that you are modifying the internal state of the class, not just some local variable